src/sections/dataset/dataset-files/useFiles.tsx

```diff
@@ -40,25 +38,31 @@ export function useFiles(
   useEffect(() => {
     setIsLoading(true)
 
-    if (filesCountInfo.total !== 0 && paginationInfo?.totalFiles !== 0) {
-      getFilesByDatasetPersistentId(
-        filesRepository,
-        datasetPersistentId,
-        datasetVersion,
-        paginationInfo,
-        criteria
-      )
-        .then((files: File[]) => {
-          setFiles(files)
-          return files
-        })
-        .then((files: File[]) =>
-          fetchFilesPermission(FilePermission.DOWNLOAD_FILE, files).then(() => setIsLoading(false))
+    if (filesCountInfo) {
+      if (filesCountInfo.total === 0) {
+        setIsLoading(false)
+      } else {
+        getFilesByDatasetPersistentId(
+          filesRepository,
+          datasetPersistentId,
+          datasetVersion,
+          paginationInfo,
+          criteria
         )
-        .catch((error) => {
-          console.error('There was an error getting the files', error)
-          setIsLoading(false)
-        })
+          .then((files: File[]) => {
+            setFiles(files)
+            return files
+          })
+          .then((files: File[]) =>
+            fetchFilesPermission(FilePermission.DOWNLOAD_FILE, files).then(() =>
+              setIsLoading(false)
+            )
+          )
+          .catch((error) => {
+            console.error('There was an error getting the files', error)
```

Comment:
The linter doesn't allow me to not have a catch statement then if I have a catch statement without a console.error the error becomes invisible. I could remove the console.error as you suggest if I didn't have a catch statement because then the error appears in the console. 

I mean, if we had an error handling and logging strategy then the console.error wouldn't be necessary, but until then this is what we have

It was a different case in the other PR because there was a throw Error in those catch statements so the error was being thrown elsewhere instead of being catched and silenced

Let me know if you come up with another solution, but not being able to see the errors is not an option

In any case I'll review the code to check if there is any console.log that I used for debugging and I forgot to remove